src/services/wallets/torus.ts

```diff
@@ -0,0 +1,64 @@
+import { HexString } from "../../utilities/types";
+import web3Torus, { BuildEnvironment } from "./tweb3";
+
+export const enableTorus = async (
+  buildEnv?: BuildEnvironment
+): Promise<void> => {
+  await web3Torus.initialize(buildEnv || "testing");
+};
+
+export const isInitialized = (): boolean => {
+  return web3Torus.initialized;
+};
+
+interface UserInfo {
+  email: string;
+  name: string;
+  profileImage: string;
+  verifier: string;
+  verifierId: string;
+}
+
+export const getUserInfo = async (): Promise<UserInfo> => {
+  const userInfo = await web3Torus.torus?.getUserInfo(
+    "This site is requesting your information"
+  );
+  if (!userInfo) throw new Error("Unable to find user's info");
+  return userInfo;
+};
+
+export const logout = (): void => {
+  web3Torus.torus?.cleanUp().then(() => {
+    sessionStorage.setItem("pageUsingTorus", "false");
+  });
+};
+
+type VerifierTypes = "google" | "reddit" | "discord";
+type AddressType = undefined | string | { address: string };
+export const getPublicAddress = async (
+  verifier: VerifierTypes,
+  verifierId: string
+): Promise<AddressType> => {
+  return await web3Torus.torus?.getPublicAddress({
+    verifier,
+    verifierId,
+    isExtended: true,
+  });
+};
+
+export const getWalletAddress = async (): Promise<string> => {
+  const { verifier, verifierId } = await getUserInfo();
+  const walletObject = await getPublicAddress(
```

Comment:
the return value of this is an object or a string?
why not call this publicAddress?

Reply:
It returns the wallet address of the user.

Since we have both a social identity address generated by the DSNP and a blockchain generated user/wallet address which are both public we need to be more specific. So we have opted to use `socialAddress` and `walletAddress` respectively to refer to these two different strings of characters.

Reply:
I think writing address is ok here. WalletObject is more confusing.